src/soundex.js

```diff
@@ -32,8 +26,25 @@ function generateSoundex(name) {
     return soundex.join('');
 }
 
+function iterateChars(name, soundex, prevCode) {
+    for (let i = 1; i < name.length && soundex.length < 4; i++) {
+        let code = getSoundexCode(name[i]);
+        comparePrevAndCurrentCode(code, prevCode, soundex);
+        prevCode = code;
+    }
+}
+
+function comparePrevAndCurrentCode(code, prevCode, soundex) {
+    if (code !== '0' && code !== prevCode) {
+        soundex.push(code);
+    }
+}
+
 module.exports = {
     getSoundexCode,
```

Comment:
Dont export getSoundexCode because its not a part of public interface of the module